### PR TITLE
Improve logging with objects

### PR DESCRIPTION
**Pull request by andrewballantyne**
Closes: #1826

## Description

Improve logging so we can filter through the full error object as we apply scenarios.

## How Has This Been Tested?

Startup in dev mode as a regular user.

## Test Impact

None

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: backend/src/plugins/kube.ts [hunk 1]
```diff
@@ -46,7 +46,10 @@ export default fp(async (fastify: FastifyInstance) => {
     );
     clusterID = (clusterVersion.body as { spec: { clusterID: string } }).spec.clusterID;
   } catch (e) {
-    fastify.log.error(`Failed to retrieve cluster id: ${e.response?.body?.message || e.message}.`);
+    fastify.log.error(
+      e,
+      `Failed to retrieve cluster id ${e.response?.body?.message || e.message}.`,
```
Comment (christianvogt) [inline backend/src/plugins/kube.ts:51]: Unintentionally removed the `:` ?
```suggestion
      `Failed to retrieve cluster id: ${e.response?.body?.message || e.message}.`,
```

File: backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts [hunk 2]
```diff
@@ -137,7 +135,7 @@ export const getClusterSettings = async (
       clusterSettings.userTrackingEnabled =
         segmentEnabledRes.body.data.segmentKeyEnabled === 'true';
     } catch (e) {
-      fastify.log.error('Error retrieving segment key enabled: ' + e.toString());
+      fastify.log.error(e, 'Error retrieving segment key enabled');
```
Comment (christianvogt) [inline backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts:138]: Make messages consistent with or without ending period.

File: backend/src/plugins/kube.ts (hunk 1 above)
Comment (andrewballantyne) [reply backend/src/plugins/kube.ts:51]: Accident -- I was cutting and pasting, didn't review closely. Will fix.

**Pull request merged**